gwpy/frequencyseries/frequencyseries.py

```diff
@@ -349,16 +347,24 @@ def from_lal(cls, lalfs, copy=True):
         """Generate a new `FrequencySeries` from a LAL `FrequencySeries`
         of any type.
         """
+        # convert units
         from ..utils.lal import from_lal_unit
         try:
             unit = from_lal_unit(lalfs.sampleUnits)
         except TypeError:
             unit = None
-        channel = Channel(lalfs.name, unit=unit,
-                          dtype=lalfs.data.data.dtype)
-        return cls(lalfs.data.data, channel=channel, f0=lalfs.f0,
-                   df=lalfs.deltaF, epoch=float(lalfs.epoch),
-                   dtype=lalfs.data.data.dtype, copy=copy)
+
+        # create a new series
+        return cls(
+            lalfs.data.data,
+            name=lalfs.name or None,
+            unit=unit,
+            f0=lalfs.f0,
+            df=lalfs.deltaF,
+            epoch=lalfs.epoch,
+            channel=None,
+            copy=copy,
```

Comment:
Just noting two other changes here for you to check that it's fine:
1. `float(lalfs.epoch)` ==> `lalfs.epoch` (dropping the cast to `float`)
2. no longer using the `dtype` optional argument. Did you mean to drop this?

Reply:
Both of these are deliberate changes:

1. casting a `Quantity` to float is unnecessary in this case, and has the potential to introduce errors (since it implicitly drops the physical units)
2. dtype is redundant with passing the actual data, so is also unnecessary

Reply:
@eagoetz, is my answer satisfactory?

Reply:
@duncanmmacleod Looks good, thanks! I'll mark resolved